client/jetpack-connect/authorize.js

```diff
@@ -610,7 +610,13 @@ export class JetpackAuthorize extends Component {
 		return (
 			<LoggedOutFormLinks>
 				{ this.isWaitingForConfirmation() ? backToWpAdminLink : null }
-				<LoggedOutFormLinkItem href={ login( { redirectTo: window.location.href } ) }>
+				<LoggedOutFormLinkItem
+					href={ login( {
+						isJetpack: true,
+						isNative: config.isEnabled( 'login/native-login-links' ),
```

Comment:
So we weren't using the Calypso-native login before for this link?

Reply:
Nope!

It was pointing to `https://wordpress.com/wp-login.php?redirect_to=…`. Of course, there's a redirection in place to `/log-in` so it's easy to miss.

I'm not sure why this option exists and or doesn't default to true 🤷‍♂️ 
It certainly seems like the config check could be internal to the `login` function.

Reply:
> It certainly seems like the config check could be internal to the login function.

Yeah, my thoughts exactly.

Reply:
Nice catch with this one!